Library/Formula/reposurgeon.rb

```diff
@@ -1,18 +1,39 @@
-require 'formula'
-
 class Reposurgeon < Formula
-  homepage 'http://www.catb.org/esr/reposurgeon/'
-  url 'http://www.catb.org/~esr/reposurgeon/reposurgeon-3.10.tar.gz'
-  sha256 'cfb78271b31ab50a73775a4a1b211dc8b48af76ac092676a226c9e61f4a1fb44'
+  homepage "http://www.catb.org/esr/reposurgeon/"
+  url "http://www.catb.org/~esr/reposurgeon/reposurgeon-3.21.tar.gz"
+  sha256 "637f1450b235f0a08d6f850e565a60b357607ce3a95dcbaea15425947fd002b9"
+
+  head "git://thyrsus.com/repositories/reposurgeon.git"
 
-  depends_on 'asciidoc'
-  depends_on 'xmlto'
+  depends_on "asciidoc" => :build
+  depends_on "xmlto" => :build
 
   def install
-    ENV['XML_CATALOG_FILES'] = "#{etc}/xml/catalog"
-    system "make"
-    tools = %w{reposurgeon repopuller repodiffer}
-    bin.install tools
-    man1.install tools.map { |m| "#{m}.1" }
+    ENV["XML_CATALOG_FILES"] = "#{etc}/xml/catalog"
+    system "make", "install", "prefix=#{prefix}"
+    (share/"emacs/site-lisp").install "reposurgeon-mode.el"
+  end
+
+  def caveats; <<-EOS.undent
+    An Emacs mode has been installed in #{HOMEBREW_PREFIX}/share/emacs/site-lisp
+    Add it to your load-path to use reposurgeon-mode.el
+    EOS
+  end
+
+  test do
+    (testpath/".gitconfig").write <<-EOS.undent
+      [user]
+        name = Real Person
+        email = notacat@hotmail.cat
+      EOS
+    system "git", "init"
+
+    require "open3"
```

Comment:
I don't think it allows commands to be piped in.  When I try `assert_match "found git repository", pipe_output("#{bin}/reposurgeon", "read")` I get this:
```
🐙  brew test -v reposurgeon
Testing reposurgeon
==> git init
Initialized empty Git repository in /private/tmp/reposurgeon20150510-795-1xv6t0v/.git/
==> /usr/local/Cellar/reposurgeon/3.21/bin/reposurgeon
reposurgeon: fatal exception in popen_or_die.
reposurgeon:
```